### DISCUSSION TITLE
Can't switch GPU in standard GPU mode

### DISCUSSION THREAD

**Discussion by CNXZY** [seerge/g-helper#653] [Q&A] [answered]
When using Armoury Crate, I can set it to standard mode, allowing me to freely switch between iGPU and dGPU using the NVIDIA Control Panel without restarting. However, using this software results in a frozen screen, and even when playing games, the iGPU power consumption is limited to 20 watts.
I'm Using System Model ASUS TUF Gaming A15 FA507XV.

**Comment by seerge** [comment]
Standard mode in g-helper also enables dGPU (letting windows decide what gpu to use), same as in AC

Given that you don’t provide any debug info I won’t be even guessing what happens on your side.

**Reply by CNXZY** [reply]
I'm sorry, but I don't know how to provide debug info. Could you tell me how to do it? This problem has troubled me for a long time, if I can't solve it, I can only use the  Armory Crate.😭

**Reply by seerge** [reply]
You can post log from %appdata%/ghelper

**Reply by CNXZY** [reply]
Here is yesterday's log file, and I hope this file can help resolve the issue.
[log.txt](https://github.com/seerge/g-helper/files/11812184/log.txt)

**Comment by seerge** [comment]
From you log i see that the last time (and the only time) you have disabled eco mode (i.e. set to standradrd) went fine :

```
2023/6/20 13:43:24: Running eco command 0
2023/6/20 13:43:27: GPUEco = 0 : OK
2023/6/20 13:43:27: Eco flag : 0
```

Since then your GPU is detected and seem to be available 

```
2023/6/20 16:57:45: NVIDIA GeForce RTX 4060 Laptop GPU
```

What exactly is not working ?

Just in case you can turn dGPU on off with a simple powershell command as admin as well

```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090020, 1)  disable GPU (eco)

(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00090020, 0)  enable GPU (standard)

```

**Reply by CNXZY** [reply]
<img width="749" alt="Snipaste_2023-06-21_19-54-07" src="https://github.com/seerge/g-helper/assets/126311016/8a5ab1c7-1329-4074-9014-a0ed011dc95d">
My optimus  setting is on "Optimus", and can be switched to "Automatically Select," but it cannot be set to "Nvidia GPU Only."

**Reply by seerge** [reply]
@CNXZY that's on a nvidia / driver level (optimus is part of that) :) g-helper doesn't interfere there anyhow. 
If AC works (by some reason) better - you can keep using it w/o any problems of course.

**Reply by CNXZY** [reply]
Later, I decided to restart, but the screen got stuck at the PIN input interface and didn't move. After another restart, it returned to normal.
It's log.
[log.txt](https://github.com/seerge/g-helper/files/11817446/log.txt)

**Reply by seerge** [reply, marked as answer]
It looks more like some issues with your windows then , if PIN is not working :) 

G-helper **is not** a driver, not your windows and not a bios. It's just an app that lets you select predefined (in bios) performance modes / fan curves / etc. Same as AC.

**Reply by CNXZY** [reply]
Thank you for your help. This appears to be an error on the part of ASUS or NVIDIA.I hope that this bug can be fixed as soon as possible in the future.

**Reply by seerge** [reply]
Try to do a clean install of NVidia driver for start to see if it helps.

**Comment by Tanvir-GitHub** [comment]
I had the same problem on my ASUS ROG STRIX G15. Try uninstalling armoury crate and then reinstalling the software to solve the bug issue. If you still can't find the GPU on the device manager try uninstalling the GPU drivers completely then install the driver using custom installation and proper clean installation. See youtube if you have any doubts.
